src/library_manager/lib_manager.c

```diff
@@ -611,89 +615,128 @@ int lib_manager_load_library(uint32_t dma_id, uint32_t lib_id)
 		.block_count = 1,
 		.head_block = &dma_block_cfg,
 	};
-	struct lib_manager_dma_ext dma_ext;
-	uint32_t addr_align;
-	int ret, ret2;
-	void __sparse_cache *man_tmp_buffer = NULL;
+	int ret;
 
-	if (!lib_id || lib_id >= LIB_MANAGER_MAX_LIBS) {
-		tr_err(&lib_manager_tr,
-		       "lib_manager_load_library(): invalid lib_id: %u", lib_id);
-		return -EINVAL;
-	}
+	if (_ext_lib->runtime_data)
+		return 0;
 
-	lib_manager_init();
+	dma_ext = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+			  sizeof(*dma_ext));
+	if (!dma_ext)
+		return -ENOMEM;
 
-	ret = lib_manager_dma_init(&dma_ext, dma_id);
+	ret = lib_manager_dma_init(dma_ext, dma_id);
 	if (ret < 0)
-		return ret;
+		goto err_dma_init;
 
 #if CONFIG_ZEPHYR_NATIVE_DRIVERS
-	ret = dma_get_attribute(dma_ext.dma->z_dev, DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT,
-				&addr_align);
+	ret = dma_get_attribute(dma_ext->dma->z_dev, DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT,
+				&dma_ext->addr_align);
 #else
-	ret = dma_get_attribute_legacy(dma_ext.dma, DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT,
-				       &addr_align);
+	ret = dma_get_attribute_legacy(dma_ext->dma, DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT,
+				       &dma_ext->addr_align);
 #endif
 	if (ret < 0)
-		goto e_init;
-
-	/* allocate temporary manifest buffer */
-	man_tmp_buffer = (__sparse_force void __sparse_cache *)rballoc_align(0, SOF_MEM_CAPS_DMA,
-								MAN_MAX_SIZE_V1_8, addr_align);
-	if (!man_tmp_buffer) {
-		ret = -ENOMEM;
-		goto e_init;
-	}
+		goto err_dma_init;
```

Comment:
Not blocking this PR, but I think we can drop the non-native drivers support. This would be nice to have, but I don't think the code really supports non-native buld. E.g. we have ifdefs around dma_get_attribute, but not around dma_start

Reply:
removed